File: src/dataset/cpp/food_chain_Exercism_MaxyMoos/claude/ClaudeSonnet4_food_chain_v1.cpp
#include "food_chain.h"

using namespace std;

////////////////////////
// ANIMAL UTIL CLASS
////////////////////////
animal::animal(string name, string secondLine)
    : name(std::move(name)), secondLine(std::move(secondLine)), optional(this->name)
{
}

animal::animal(string name, string secondLine, string optional)
    : name(std::move(name)), secondLine(std::move(secondLine)), optional(optional.empty() ? this->name : std::move(optional))
{
}

static const animal animals[8]{
    {"fly", ""},
    {"spider", "It wriggled and jiggled and tickled inside her.\n", "spider that wriggled and jiggled and tickled inside her"},
    {"bird", "How absurd to swallow a bird!\n"},
    {"cat", "Imagine that, to swallow a cat!\n"},
    {"dog", "What a hog, to swallow a dog!\n"},
    {"goat", "Just opened her throat and swallowed a goat!\n"},
    {"cow", "I don't know how she swallowed a cow!\n"},
    {"horse", ""}
};

///////////////////////////
// MAIN COURSE IS HERE
///////////////////////////

string food_chain::verse(int index)
{
    const auto& animal = animals[index-1];
    string result;
    result.reserve(256); // Pre-allocate reasonable capacity
    
    result += food_chain::firstLine;
    result += animal.name;
    result += ".\n";
    result += animal.secondLine;
    
    if (index == 8) {
        result += "She's dead, of course!\n";
    } else {
        for (int i = index-1; i > 0; --i) {
            result += "She swallowed the ";
            result += animals[i].name;
            result += " to catch the ";
            result += animals[i-1].optional;
            result += ".\n";
        }
        result += food_chain::finalStr;
    }
    return result;
}

string food_chain::verses(int lowIndex, int highIndex)
{
    string result;
    result.reserve(1024 * (highIndex - lowIndex + 1)); // Pre-allocate capacity
    
    for (int i = lowIndex; i <= highIndex; ++i) {
        result += food_chain::verse(i);
        result += "\n";
    }
    return result;
}

string food_chain::sing()
{
    return food_chain::verses(1, 8);
}

File: src/dataset/typescript/food-chain_Exercism-typescript-shybyte/claude/ClaudeSonnet4_food-chain_v1.ts
const FOOD_CHAIN = [
    { animal: 'fly', sentence: "I don't know why she swallowed the fly. Perhaps she'll die.", optionalDescription: '' },
    { animal: 'spider', sentence: 'It wriggled and jiggled and tickled inside her.', optionalDescription: ' that wriggled and jiggled and tickled inside her' },
    { animal: 'bird', sentence: 'How absurd to swallow a bird!', optionalDescription: '' },
    { animal: 'cat', sentence: 'Imagine that, to swallow a cat!', optionalDescription: '' },
    { animal: 'dog', sentence: 'What a hog, to swallow a dog!', optionalDescription: '' },
    { animal: 'goat', sentence: 'Just opened her throat and swallowed a goat!', optionalDescription: '' },
    { animal: 'cow', sentence: "I don't know how she swallowed a cow!", optionalDescription: '' },
    { animal: 'horse', sentence: "She's dead, of course!", optionalDescription: '' }
] as const

function firstLineInVerse(animal: string) {
    return `I know an old lady who swallowed a ${animal}.`
}

function verse(verseNumber: number) {
    const lines: string[] = []
    
    if (verseNumber === 8) {
        const {animal, sentence} = FOOD_CHAIN[7]
        lines.push(firstLineInVerse(animal), sentence)
        return lines.join('\n') + '\n'
    }
    
    const {animal: firstAnimal, sentence: firstSentence} = FOOD_CHAIN[verseNumber - 1]
    lines.push(firstLineInVerse(firstAnimal), firstSentence)
    
    for (let i = verseNumber - 2; i >= 0; i--) {
        const current = FOOD_CHAIN[i]
        const previous = FOOD_CHAIN[i + 1]
        lines.push(`She swallowed the ${previous.animal} to catch the ${current.animal}${current.optionalDescription}.`)
    }
    
    if (verseNumber > 1) {
        lines.push(FOOD_CHAIN[0].sentence)
    }
    
    return lines.join('\n') + '\n'
}

function verses(minVerse = 1, maxVerse = 8) {
    let result = ''
    for (let i = minVerse; i <= maxVerse; i++) {
        result += verse(i)
        if (i < maxVerse) result += '\n'
    }
    return result
}

export default {verse, verses}